diff -Nru libnet.orig/src/Makefile.am libnet/src/Makefile.am
--- libnet.orig/src/Makefile.am	2003-10-13 23:08:30.000000000 +0200
+++ libnet/src/Makefile.am	2003-11-27 12:00:33.460591752 +0100
@@ -56,7 +56,7 @@
 			libnet_version.c \
 			libnet_write.c
 
-EXTRA_libnet_a_SOURCES = libnet_link_bpf.c \
+EXTRA_DIST = libnet_link_bpf.c \
 			libnet_link_dlpi.c \
 			libnet_link_linux.c \
 			libnet_link_nit.c \
