diff -Nru libnet.orig/configure.in libnet/configure.in
--- libnet.orig/configure.in	2003-10-13 23:08:30.000000000 +0200
+++ libnet/configure.in	2003-11-27 12:16:12.126892784 +0100
@@ -20,7 +20,7 @@
 dnl
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 AC_CHECK_HEADERS(sys/sockio.h)
 
diff -Nru libnet.orig/sample/Makefile.am libnet/sample/Makefile.am
--- libnet.orig/sample/Makefile.am	2003-11-01 18:09:21.000000000 +0100
+++ libnet/sample/Makefile.am	2003-11-27 12:16:28.018476896 +0100
@@ -55,4 +55,4 @@
 bgp4_notification_SOURCES = bgp4_notification.c
 gre_SOURCES		= gre.c
 
-LDADD = $(top_srcdir)/src/libnet.a
+LDADD = $(top_srcdir)/src/libnet.la
diff -Nru libnet.orig/src/Makefile.am libnet/src/Makefile.am
--- libnet.orig/src/Makefile.am	2003-10-13 23:08:30.000000000 +0200
+++ libnet/src/Makefile.am	2003-11-27 12:19:10.180824472 +0100
@@ -8,9 +8,9 @@
 
 include $(top_srcdir)/Makefile.am.common
 
-lib_LIBRARIES = libnet.a
+lib_LTLIBRARIES = libnet.la
 
-libnet_a_SOURCES = libnet_asn1.c \
+libnet_la_SOURCES = libnet_asn1.c \
 			libnet_build_802.1q.c \
 			libnet_build_802.1x.c \
 			libnet_build_802.2.c \
@@ -66,5 +66,6 @@
 			libnet_link_snoop.c \
 			libnet_link_win32.c
 
-libnet_a_LIBADD = @LIBOBJS@
+libnet_la_LIBADD = @LIBOBJS@
+libnet_la_LDFLAGS = -version-info 2:0:0
 
