--- libnet/src/libnet_if_addr.c.bak	2008-11-03 17:06:55.493985477 +0100
+++ libnet/src/libnet_if_addr.c	2008-11-03 17:07:07.317325455 +0100
@@ -141,6 +141,7 @@
 	snprintf(errbuf, LIBNET_ERRBUF_SIZE,
                 "%s(): fopen(proc_dev_file) failed: %s\n",  __func__,
                 strerror(errno));
+	close(fd);
 	return (-1);
     }
 #endif
@@ -251,7 +263,9 @@
 #endif
 
     } /* while|for */
-	
+
+    close(fd);
+
 #ifdef HAVE_LINUX_PROCFS
     if (ferror(fp))
     {
