--- libnet/acinclude.m4~	2004-01-16 00:53:06.000000000 +0100
+++ libnet/acinclude.m4	2009-08-28 21:58:51.368425567 +0200
@@ -12,18 +12,9 @@
 dnl results:    HAVE_LINUX_PROCFS
 dnl
 AC_DEFUN(AC_LIBNET_LINUX_PROCFS,
-    [AC_MSG_CHECKING(for Linux proc filesystem)
-    AC_CACHE_VAL(ac_cv_libnet_linux_procfs,
-        if test "x`cat /proc/sys/kernel/ostype 2>&-`" = "xLinux" ; then
-            ac_cv_libnet_linux_procfs=yes
-        else
-            ac_cv_libnet_linux_procfs=no
-        fi)
-    AC_MSG_RESULT($ac_cv_libnet_linux_procfs)
-    if test $ac_cv_libnet_linux_procfs = yes ; then
-        AC_DEFINE(HAVE_LINUX_PROCFS, 1,
+    [AC_DEFINE(HAVE_LINUX_PROCFS, 1,
                   [Define if you have the Linux /proc filesystem.])
-    fi])
+    ])
 
 dnl
 dnl Checks to see if this linux kernel has a working PF_PACKET
